jenkins/job-configs/kubernetes-jenkins/kubernetes-build.yaml

```diff
@@ -43,17 +44,31 @@
 - project:
     name: kubernetes-builds
     build:
-        - 'build':
-            branch: 'master'
-            timeout: 50
-        - 'build-1.0':
-            branch: 'release-1.0'
-            timeout: 30
-        - 'build-1.1':
-            branch: 'release-1.1'
-            timeout: 30
-        - 'build-1.2':
-            branch: 'release-1.2'
-            timeout: 30
+      - 'build':
+          url: 'https://github.com/kubernetes/kubernetes'
+          branch: 'master'
+          timeout: 50
+      - 'build-1.0':
+          url: 'https://github.com/kubernetes/kubernetes'
+          branch: 'release-1.0'
+          timeout: 30
+      - 'build-1.1':
+          url: 'https://github.com/kubernetes/kubernetes'
+          branch: 'release-1.1'
+          timeout: 30
+      - 'build-1.2':
+          url: 'https://github.com/kubernetes/kubernetes'
+          branch: 'release-1.2'
+          timeout: 30
+
+      - 'federation-build':
+          branch: 'federation-e2e'
+          url: 'https://github.com/colhom/kubernetes'
```

Comment:
This is for the "folks at home" that want to run their own Jenkins against their own branches, which is basically what I was doing as part of development for kubernetes/kubernetes#25125. It's nice to be able to cleanly parameterize a new job which pulls off a custom branch without mutilating the project template.